DNN Platform/Library/WebControls/WebControlBase.cs

```diff
@@ -1,95 +1,123 @@
-            }
-        }
-
-        [Browsable(false)]
-        [DesignerSerializationVisibility(DesignerSerializationVisibility.Hidden)]
-        public PortalSettings PortalSettings
-        {
-            get
-            {
-                return PortalController.Instance.GetCurrentPortalSettings();
-            }
-        }
-
-        public abstract string HtmlOutput { get; }
-
-        [Obsolete("There is no longer the concept of an Admin Page.  All pages are controlled by Permissions. Scheduled removal in v11.0.0.", true)]
-        public bool IsAdminMenu
-        {
-            get
-            {
-                bool _IsAdmin = false;
-                return _IsAdmin;
-            }
-        }
-
-        public string Theme
-        {
-            get
-            {
-                return this._theme;
-            }
-
-            set
-            {
-                this._theme = value;
-            }
-        }
-
-        public string StyleSheetUrl
-        {
-            get
-            {
-                if (this._styleSheetUrl.StartsWith("~"))
-                {
-                    return Globals.ResolveUrl(this._styleSheetUrl);
-                }
-                else
-                {
-                    return this._styleSheetUrl;
-                }
-            }
-
-            set
-            {
-                this._styleSheetUrl = value;
-            }
-        }
-
-        protected override void RenderContents(HtmlTextWriter output)
-        {
-            output.Write(this.HtmlOutput);
-        }
-    }
-}
+    /// <summary>
+    /// Base class for Dnn specific WebControls.
+    /// </summary>
+    public abstract class WebControlBase : WebControl
+    {
+        private string styleSheetUrl = string.Empty;
+        private string theme = string.Empty;
+
+        /// <summary>
+        /// Gets a url to the web control resources folder.
+        /// </summary>
+        public string ResourcesFolderUrl
+        {
+            get
+            {
+                return Globals.ResolveUrl("~/Resources/");
+            }
+        }
+
+        /// <summary>
+        /// Gets a value indicating whether the control is currently in the host menu.
+        /// </summary>
+        [Obsolete("There is no longer the concept of aa Host Page.  All pages are controlled by Permissions. Scheduled removal in v11.0.0.")]
+        public bool IsHostMenu
+        {
+            get
+            {
+                return Globals.IsHostTab(this.PortalSettings.ActiveTab.TabID);
+            }
+        }
+
+        /// <summary>
+        /// Gets the portal settings for the portal where the WebControl is displayed.
+        /// </summary>
+        [Browsable(false)]
+        [DesignerSerializationVisibility(DesignerSerializationVisibility.Hidden)]
+        public PortalSettings PortalSettings
+        {
+            get
+            {
+                return (PortalSettings)PortalController.Instance.GetCurrentSettings();

```

Comment:
Rather than casting the interface, can we just mark this as obsolete, indicating they should call `PortalController.Instance.GetCurrentSettings()` directly?

Reply:
Ok, I have deprecated PortalSettings in here, changed our usage of it in this file to TabController.CurrentPage but I left the cast in so we can have this file with 0 warnings, does that work out ?

Reply:
While the cast should practically work for the foreseeable future, it's creating a dependency that the interface usage is attempting to break (`IPortalSettings` doesn't _have to_ always be a `PortalSettings` instance).  I'd prefer to suppress the warning in this instance (in addition to deprecating).

```suggestion
#pragma warning disable 612, 618 // GetCurrentPortalSettings is obsolete
                return PortalController.Instance.GetCurrentPortalSettings();
#pragma warning restore 612, 618 
```

Reply:
Works for me